app/src/main/java/com/github/wnder/TakePictureActivity.java

```diff
@@ -0,0 +1,131 @@
+package com.github.wnder;
+
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.content.FileProvider;
+
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.net.Uri;
+import android.os.Bundle;
+import android.os.Environment;
+import android.provider.MediaStore;
+import android.view.View;
+import android.widget.Button;
+import android.widget.ImageView;
+
+import java.io.File;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class TakePictureActivity extends AppCompatActivity {
+    private Button takePictureButton;
+    private Button pictureConfirmButton;
+    private ImageView picture;
+    private final int TAKE_PHOTO = 0;
+    private String currentPhotoPath;
+    private String imageFileName;
+    private Bitmap currentBitmap;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_take_picture);
+        picture = findViewById(R.id.imageFromCamera);
+
+        takePictureButton = findViewById(R.id.takePictureButton);
+        takePictureButton.setOnClickListener(new View.OnClickListener() {
```

Comment:
For both onClickListener as they are functional interfaces you could use lamba to make the code a bit clearer (and shorter)